.github/workflows/ci.yml

```diff
@@ -64,6 +64,17 @@ jobs:
         with:
           crate: cargo-sort
           version: "^1.0"
+      - name: check proto build up-to-date
+        run: |
+          cargo build --features prost-build          
+          if git diff proto/; then
```

Comment:
Why do we want to check if `proto/` changed or not?

Reply:
Oh yeah, but the `else` branch is still:

```
git diff --exit-code src/api.rs
```

Doesn't having only this suffice?

Reply:
yeah.. gotta test it out, i just wrote it lol

Reply:
no i meant like, you open a PR to upgrade protos and lets say you forgot to regen src/api.rs but git diff with main is ok, means all checks pass. i wanted to consider that but the bash code may not be enough/right

Reply:
Yup, the git diff for `api.rs` should be enough for everything.

Reply:
```
git diff --exit-code src/api.rs
```

Won't be `0` if any updates are uncommitted

Reply:
ah ok! thanks for reminding me of that